### PR TITLE
CLN/BUG: Clean/Simplify _wrap_applied_output

### PR DESCRIPTION
**Pull request by rhshadrach**
- [ ] closes #xxxx
- [x] tests added / passed
- [x] passes `black pandas`
- [x] passes `git diff upstream/master -u -- "*.py" | flake8 --diff`
- [x] whatsnew entry

First part of #35412. I could not find a relevant issue for the test change/whatsnew update.

### CODE REVIEW

**Comment by rhshadrach** [comment]
cc @WillAyd @jbrockmendel @jreback 

mac timed out, but otherwise green.

**Review by WillAyd — APPROVED** [review]
lgtm when green

**Comment by rhshadrach** [comment]
Thanks @WillAyd for restarting the test. Passing now.

File: pandas/core/groupby/generic.py
Comment (jreback) [inline pandas/core/groupby/generic.py:1278]: can you give a comment here on what we are doing (as the original context i think is not htere anymore)

**Review by jreback — CHANGES_REQUESTED** [review]
nice! minor comment. can you run the benchmarks for groupby and report the results to make sure no regressions. ping on green.

**Comment by rhshadrach** [comment]
@jreback - comment added; only failures are due to pyarrow.

File: pandas/core/groupby/generic.py
Comment (jreback) [reply pandas/core/groupby/generic.py:1278]: can you comment here

**Review by jreback — CHANGES_REQUESTED** [review]
small comment, pls rebase and ping on green.

File: pandas/core/groupby/generic.py
Comment (rhshadrach) [reply pandas/core/groupby/generic.py:1278]: I added the comment

    # Combine values
    # vstack+constructor is faster than concat and handles MI-columns

above the `np.vstack` line; I can't tell if you're saying this is insufficient or if you're looking for a comment below this line.

File: doc/source/whatsnew/v1.2.0.rst
Comment (jreback) [inline doc/source/whatsnew/v1.2.0.rst:254]: can you add the issue number here (in this case use this PR number as there isn't an issue)

**Review by jreback — APPROVED** [review]
minor doc-comment, after fix, merge on green.

**Comment by rhshadrach** [comment]
Thanks @jreback, merged.

**Comment by jbrockmendel** [comment]
@rhshadrach as a rule of thumb, we try not to merge our own PRs

**Comment by simonjayhawkins** [comment]
@meeseeksdev backport 1.1.x

This will fail but give instructions

**Pull request merged**
